### PR TITLE
feat: add class to generate pileup over AlignedSegments

### PR DESCRIPTION
**Pull request by nh13**
Add new IteratorColumnRecords class that generates pileup columns from a
collection of AlignedSegment objects using htslib's push-based pileup API
(bam_plp_push/bam_plp64_next).

Key features:
- Accepts any iterable of AlignedSegments (requires coordinate-sorted order)
- Supports optional reference sequence (fastafile parameter)
- Includes add_reference(), has_reference(), and seq_len property
- Configurable min_base_quality parameter
- Uses 64-bit position types (hts_pos_t) for extended chromosome support

Implementation notes:
- Uses bam_plp_push/bam_plp64_next instead of callback-based approach
- Records consumed during initialization for push-based API
- Includes required NULL push to signal end-of-input
- Leverages 64-bit APIs (bam_plp64_next, faidx_fetch_seq64) from PR #1362
- Uses opaque bam_plp_s struct (no direct field access needed)

Testing:
- 12 new tests covering reference support, edge cases, and parameters
- Documented known limitation: minor depth differences vs samtools mpileup
  due to push-based vs pull-based filtering differences

Changes:
- Add IteratorColumnRecords class in pysam/libcalignmentfile.pyx
- Update to use 64-bit pileup APIs (bam_plp64_next, faidx_fetch_seq64)
- Add type stub in pysam/libcalignmentfile.pyi
- Add parameterized to test dependencies for parameterized tests
- Update CI workflows to install parameterized package

Closes https://github.com/pysam-developers/pysam/issues/1352

### CODE REVIEW

File: pysam/libcalignmentfile.pxd [hunk 1]
```diff
@@ -144,6 +143,19 @@ cdef class IteratorColumnAllRefs(IteratorColumn):
 cdef class IteratorColumnAll(IteratorColumn):
     pass
 
+cdef class IteratorColumnRecords:
```
Comment (nh13) [inline pysam/libcalignmentfile.pxd:146]: this cannot extend `IteratorColumn` since `IteratorColumn` requires a SAM file.

**Comment by nh13** [comment]
@jmarshall https://github.com/pysam-developers/pysam/pull/1353

**Comment by jmarshall** [comment]
I've now finally finished up and merged #1362. Can I interest you in rebasing this onto current **master** and ideally updating it to use `bam_plp64_next()` et al?

**Comment by nh13** [comment]
@jmarshall done!  Ready for re-review!

**Comment by nh13** [comment]
@jmarshall bump!

**Comment by jmarshall** [comment]
I know :smile: — It's on the short list!

**Comment by clintval** [comment]
I find myself wanting this feature often and having to home-roll every time.

Any hope for this PR? If it's gone stale, I can refresh!

**Comment by nh13** [comment]
I'll rebase shortly, so it's again ready for review
